service/src/main/java/org/broadinstitute/listener/relay/inspectors/SamResourceClient.java

```diff
@@ -1,34 +1,39 @@
 package org.broadinstitute.listener.relay.inspectors;
 
+import com.google.common.annotations.VisibleForTesting;
 import java.io.IOException;
 import java.time.Instant;
+import okhttp3.OkHttpClient;
 import org.broadinstitute.dsde.workbench.client.sam.ApiClient;
 import org.broadinstitute.dsde.workbench.client.sam.ApiException;
 import org.broadinstitute.dsde.workbench.client.sam.api.ResourcesApi;
+import org.broadinstitute.dsde.workbench.client.sam.api.UsersApi;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.cache.annotation.Cacheable;
 
 public class SamResourceClient {
+  private final String samUrl;
   private final String samResourceId;
   private final String samResourceType;
   private final TokenChecker tokenChecker;
-  private final ApiClient samClient;
```

Comment:
I was thinking reusing an `ApiClient` could cause thread safety issues since we're calling `setAccessToken` for every request. I changed it to instantiate an `ApiClient` for each request, but use a common thread pool. [WSM](https://github.com/DataBiosphere/terra-workspace-manager/blob/main/service/src/main/java/bio/terra/workspace/service/iam/SamService.java#L100) and [BPM](https://github.com/DataBiosphere/terra-billing-profile-manager/blob/main/service/src/main/java/bio/terra/profile/service/iam/SamService.java#L449) work the same way.